### PR TITLE
Feature/sg 13 forward pass prep fn

### PR DESCRIPTION
**Pull request by shaydeci**
Support for on-the-fly multiscaling for yolox.
Added as forward_pass_prep_fn since the synchronisation interferes when done on the data side.

### CODE REVIEW

**Review by ofrimasad — COMMENTED** [review]
initial review

File: src/super_gradients/training/datasets/datasets_utils.py [hunk 1]
```diff
@@ -531,16 +598,3 @@ def get_color_augmentation(rand_augment_config_string: str, color_jitter: tuple,
     else:  # RandAugment includes colorjitter like augmentations, both cannot be applied together.
         color_augmentation = transforms.ColorJitter(*color_jitter)
     return color_augmentation
-
-
-def worker_init_reset_seed(worker_id):
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/training/datasets/datasets_utils.py]: Accidentally, restored in https://github.com/Deci-AI/super-gradients/pull/224/commits/16e078dd8c3aed2631becaa23ba7b39cbc7caa95

File: src/super_gradients/training/datasets/datasets_utils.py [hunk 2]
```diff
@@ -23,6 +24,8 @@
 
 from super_gradients.training.utils.utils import AverageMeter
 from super_gradients.training.utils.detection_utils import DetectionVisualization
+import uuid
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/training/datasets/datasets_utils.py:27]: Was used in worker_init_reset_seed which I accidentally removed

File: src/super_gradients/training/datasets/datasets_utils.py [hunk 3]
```diff
@@ -191,6 +192,72 @@ def __call__(self, batch):
             return images, batch[1]
 
 
+class MultiscaleForwardPassPrepFunction:
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/training/datasets/datasets_utils.py]: Added in https://github.com/Deci-AI/super-gradients/pull/224/commits/93c1cf2239cc17337060c98a01bf0e470f7d18d5

**Review by ofrimasad — APPROVED** [review]
LGTM

**Pull request merged**